### PR TITLE
[INLONG-7331][Sort][Manager] Support complex type field

### LINKED ISSUE #7331: [Improve][Sort][Manager] Support complex type field
### Description

Supporting complex type field

### InLong Component

InLong Manager, InLong Sort

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by xxhannah**
### Prepare a Pull Request

- Fixes #7331 

### Motivation

Support complex type field

### Modifications

Support sub filed of complex type filed (eg: filedInfo.subFiled)

### CODE REVIEW

File: inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java [hunk 1]
```diff
@@ -635,11 +638,15 @@ private void parseFieldRelations(List<FieldInfo> fields,
...
         for (FieldInfo field : fields) {
             FieldRelation fieldRelation = fieldRelationMap.get(field.getName());
+            FormatInfo fieldFormatInfo = field.getFormatInfo();
             if (fieldRelation == null) {
-                String targetType = TableFormatUtils.deriveLogicalType(field.getFormatInfo()).asSummaryString();
+                String targetType = TableFormatUtils.deriveLogicalType(fieldFormatInfo).asSummaryString();
                 sb.append("\n    CAST(NULL as ").append(targetType).append(") AS ").append(field.format()).append(",");
                 continue;
             }
+            boolean complexType = fieldFormatInfo instanceof RowFormatInfo
```
Comment (healchow) [inline inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java:647]: Suggest extracting this check as a method into the `FieldInfoUtils`, then if we add a newly complex type, then we just modify `FieldInfoUtils` is enough, no need to modify this file.

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/FieldInfoUtils.java [hunk 2]
```diff
@@ -148,6 +148,17 @@ private static FieldInfo getFieldInfo(String fieldName, String fieldType,
...
+            case ARRAY:
+                return new ArrayFormatInfo(new StringFormatInfo());
+            case MAP:
+                return new MapFormatInfo(new StringFormatInfo(), new StringFormatInfo());
+            case STRUCT:
+                return new RowFormatInfo(new String[]{}, new FormatInfo[]{});
+            default:
+                break;
+        }
         return convertFieldFormat(type, null);
```
Comment (healchow) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/sort/util/FieldInfoUtils.java:162]:
```suggestion
        FieldType fieldType = FieldType.forName(type);
        FormatInfo formatInfo;
        switch (fieldType) {
            case ARRAY:
                formatInfo = new ArrayFormatInfo(new StringFormatInfo());
                break;
            case MAP:
                formatInfo = new MapFormatInfo(new StringFormatInfo(), new StringFormatInfo());
                break;
            case STRUCT:
                formatInfo = new RowFormatInfo(new String[]{}, new FormatInfo[]{});
                break;
            default:
                formatInfo = convertFieldFormat(type, null);
        }
        return formatInfo;
```

File: inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java (hunk 1 above)
Comment (xxhannah) [reply inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java:647]: It's a good suggestion, but ```FieldInfoUtils.java``` is in ```manager-pojo``` and ```FlinkSqlParser.java``` is in ```sort-core```, I don't think it's suitable to import ```manager-pojo``` into ```sort-core```. Any other good idea for this? ^_^

Comment (xxhannah) [reply inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java:647]: > Suggest extracting this check as a method into the `FieldInfoUtils`, then if we add a newly complex type, then we just modify `FieldInfoUtils` is enough, no need to modify this file.

It's a good suggestion, but ```FieldInfoUtils.java``` is in ```manager-pojo``` and ```FlinkSqlParser.java``` is in ```sort-core```, I don't think it's suitable to import ```manager-pojo``` into ```sort-core```. Any other good idea for this? ^_^

Comment (healchow) [reply inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/parser/impl/FlinkSqlParser.java:647]: Get it. I'll refactor it later.

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Comment by dockerzhang** [comment]
@xxhannah do we need the feature for the Dashboard module?

**Review by gong — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**